Bug:

### What is the bug?

When unsetting `target`, a segmentation fault is returned.

```
ts=2026-02-16T13:00:03.506646651Z caller=main.go:233 level=info msg="Starting application" version="(version=3.0.3, branch=release-3.0, revision=0b00ebf5d3)"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x2d04597]

goroutine 1 [running]:
github.com/grafana/mimir/pkg/api.(*API).RegisterServiceMapHandler(0x0, {0x41a1900, 0xc000e06ab0})
        /go/src/github.com/grafana/mimir/pkg/api/api.go:528 +0x57
github.com/grafana/mimir/pkg/mimir.(*Mimir).Run(0xc000ee2000)
        /go/src/github.com/grafana/mimir/pkg/mimir/mimir.go:1011 +0x405
main.main()
        /go/src/github.com/grafana/mimir/cmd/mimir/main.go:235 +0x1010
mathieu@metrics-1:~$ 
mathieu@metrics-1:~$ sudo mimir -config.file /etc/mimir/config.yml 
ts=2026-02-16T13:00:09.20334209Z caller=main.go:233 level=info msg="Starting application" version="(version=3.0.3, branch=release-3.0, revision=0b00ebf5d3)"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x2d04597]

goroutine 1 [running]:
github.com/grafana/mimir/pkg/api.(*API).RegisterServiceMapHandler(0x0, {0x41a1900, 0xc0007928f0})
        /go/src/github.com/grafana/mimir/pkg/api/api.go:528 +0x57
github.com/grafana/mimir/pkg/mimir.(*Mimir).Run(0xc000b56000)
        /go/src/github.com/grafana/mimir/pkg/mimir/mimir.go:1011 +0x405
main.main()
        /go/src/github.com/grafana/mimir/cmd/mimir/main.go:235 +0x1010
```

### How to reproduce it?

```sh
echo 'target: ""' > test.yml
mimir -config.file test.yml
```

output:

```
ts=2026-02-16T13:02:48.565339011Z caller=main.go:233 level=info msg="Starting application" version="(version=3.0.3, branch=release-3.0, revision=0b00ebf5d3)"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x2d04597]

goroutine 1 [running]:
github.com/grafana/mimir/pkg/api.(*API).RegisterServiceMapHandler(0x0, {0x41a1900, 0xc000b84ed0})
        /go/src/github.com/grafana/mimir/pkg/api/api.go:528 +0x57
github.com/grafana/mimir/pkg/mimir.(*Mimir).Run(0xc000b22000)
        /go/src/github.com/grafana/mimir/pkg/mimir/mimir.go:1011 +0x405
main.main()
        /go/src/github.com/grafana/mimir/cmd/mimir/main.go:235 +0x1010
```

### What did you think would happen?

It should return the usage or alert the user this parameter (`target`) must not be empty and should contain a list of elements.

### What was your environment?

- bare-metal
- ubuntu 20.04.5 LTS
- amd64
- mimir version 3.0.3

### Any additional context to share?

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -128,6 +128,7 @@
 * [ENHANCEMENT] Ingester: Add experimental file based Kafka consumer group offset tracking via flag `-ingest-storage.kafka.consumer-group-offset-commit-file-enforced`. #14110
 * [ENHANCEMENT] Store-gateway: Add "OOO" column to the tenant blocks page to indicate whether each block was created from out-of-order samples. #14283
 * [ENHANCEMENT] Ingester: Optimize ingestion from Kafka in clusters with mixed size tenants. #13924 #13961 #14302
+* [BUGFIX] Mimir: Fix nil pointer dereference when `-target` is set to an empty string. #14381
 * [BUGFIX] API: Fixed web UI links not respecting `-server.path-prefix` configuration. #14090
 * [BUGFIX] Distributor: Fix issue where distributors didn't send custom values of native histograms. #13849
 * [BUGFIX] Compactor: Fix potential concurrent map writes. #13053

--- a/pkg/mimir/mimir.go
+++ b/pkg/mimir/mimir.go
@@ -909,6 +909,10 @@ type Mimir struct {
 
 // New makes a new Mimir.
 func New(cfg Config, reg prometheus.Registerer) (*Mimir, error) {
+	if len(cfg.Target) == 0 {
+		return nil, fmt.Errorf("no target module specified, the -target flag must be set to a valid module (e.g. \"all\")")
+	}
+
 	if cfg.PrintConfig {
 		if err := yaml.NewEncoder(os.Stdout).Encode(&cfg); err != nil {
 			fmt.Println("Error encoding config:", err)

--- a/pkg/mimir/mimir_test.go
+++ b/pkg/mimir/mimir_test.go
@@ -1055,6 +1055,20 @@ func getHostnameAndRandomPort(t *testing.T) (string, int) {
 	return host, portNum
 }
 
+func TestNewReturnsErrorOnEmptyTarget(t *testing.T) {
+	args := []string{
+		"-target=",
+	}
+	var cfg Config
+	fs := flag.NewFlagSet("test", flag.PanicOnError)
+	cfg.RegisterFlags(fs, log.NewNopLogger())
+	require.NoError(t, fs.Parse(args))
+
+	_, err := New(cfg, prometheus.NewPedanticRegistry())
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "no target module specified")
+}
+
 type mockGrpcServiceHandler struct {
 	querierShutdownCalled atomic.Bool
 	rulerSyncRulesCalled  atomic.Bool

